css/mediaqueries/media-query-matches-in-iframe.html

```diff
@@ -0,0 +1,104 @@
+        iframe.onload = () => resolve(iframe);
+        document.body.appendChild(iframe);
+        test.add_cleanup(() => iframe.remove());
+    });
+    iframe.contentDocument.body.innerHTML = '<span>some content</span>';
+    window.preventOptimization1 = iframe.getBoundingClientRect();
+    window.preventOptimization2 = iframe.contentDocument.querySelector('span').getBoundingClientRect();
+    return iframe;
+}
+
+for (const query of ['(max-width: 150px)', '(width: 100px)', '(orientation: portrait)', '(aspect-ratio: 1/1)', '(max-aspect-ratio: 4/3)']) {
+    promise_test(async function () {
+        const iframe = await createFrameAndUpdateLayout(this);
+        const mediaQuery = iframe.contentWindow.matchMedia(query);
+        assert_true(mediaQuery.matches);
+        iframe.style.width = '200px';
+        assert_false(mediaQuery.matches);
+    }, `matchMedia('${query}').matches should update immediately`);
+}
+
+for (const query of ['(height: 100px)', '(max-height: 150px)', '(min-aspect-ratio: 3/4)']) {
+    promise_test(async function () {
+        const iframe = await createFrameAndUpdateLayout(this);
+        const mediaQuery = iframe.contentWindow.matchMedia(query);
+        assert_true(mediaQuery.matches);
+        iframe.style.height = '200px';
+        assert_false(mediaQuery.matches);
+    }, `matchMedia('${query}').matches should update immediately`);
+}
+
+for (const query of ['(min-height: 150px)', '(aspect-ratio: 1/2)']) {
+    promise_test(async function () {
+        const iframe = await createFrameAndUpdateLayout(this);
+        const mediaQuery = iframe.contentWindow.matchMedia(query);
+        assert_false(mediaQuery.matches);
+        iframe.style.height = '200px';
+        assert_true(mediaQuery.matches);
+    }, `matchMedia('${query}').matches should update immediately`);
+}
+
+for (const query of ['(min-width: 150px)', '(min-aspect-ratio: 4/3)']) {
+    promise_test(async function () {
+        const iframe = await createFrameAndUpdateLayout(this);
+        const mediaQuery = iframe.contentWindow.matchMedia(query);
+        assert_false(mediaQuery.matches);
+        iframe.style.width = '200px';
+        assert_true(mediaQuery.matches);
+    }, `matchMedia('${query}').matches should update immediately`);
+}
+
+for (const query of ['(max-width: 150px)', '(width: 100px)', '(orientation: portrait)', '(aspect-ratio: 1/1)', '(max-aspect-ratio: 4/3)']) {
+    promise_test(async function () {
+        const iframe = await createFrameAndUpdateLayout(this);
+        const mediaQuery = iframe.contentWindow.matchMedia(query);
+        let changes = 0;
+        mediaQuery.addEventListener('change', () => ++changes);
+        assert_true(mediaQuery.matches);
+        assert_equals(changes, 0);
+        iframe.style.width = '200px';
+        assert_false(mediaQuery.matches);
+        assert_equals(changes, 0);
+        await new Promise(requestAnimationFrame);
+        await new Promise(setTimeout);
```

Comment:
Is this setTimeout necessary?

Reply:
Oh, no. Will remove. I added there temporarily because Blink/Chrome fails these test cases without it.